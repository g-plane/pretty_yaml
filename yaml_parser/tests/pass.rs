use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
