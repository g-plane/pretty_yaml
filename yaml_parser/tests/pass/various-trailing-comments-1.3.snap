---
source: yaml_parser/tests/pass.rs
---
ROOT@0..157
  DOCUMENT@0..156
    BLOCK@0..156
      BLOCK_MAP@0..156
        BLOCK_MAP_ENTRY@0..20
          BLOCK_MAP_KEY@0..1
            FLOW@0..1
              PLAIN_SCALAR@0..1 "a"
          COLON@1..2 ":"
          WHITESPACE@2..3 " "
          BLOCK_MAP_VALUE@3..20
            FLOW@3..20
              DOUBLE_QUOTED_SCALAR@3..20 "\"double\n  quotes\""
        WHITESPACE@20..21 " "
        COMMENT@21..27 "# lala"
        WHITESPACE@27..28 "\n"
        BLOCK_MAP_ENTRY@28..43
          BLOCK_MAP_KEY@28..29
            FLOW@28..29
              PLAIN_SCALAR@28..29 "b"
          COLON@29..30 ":"
          WHITESPACE@30..31 " "
          BLOCK_MAP_VALUE@31..43
            FLOW@31..43
              PLAIN_SCALAR@31..43 "plain\n value"
        WHITESPACE@43..45 "  "
        COMMENT@45..51 "# lala"
        WHITESPACE@51..52 "\n"
        BLOCK_MAP_ENTRY@52..66
          BLOCK_MAP_KEY@52..53
            FLOW@52..53
              PLAIN_SCALAR@52..53 "c"
          WHITESPACE@53..55 "  "
          COLON@55..56 ":"
          WHITESPACE@56..57 " "
          COMMENT@57..62 "#lala"
          WHITESPACE@62..65 "\n  "
          BLOCK_MAP_VALUE@65..66
            FLOW@65..66
              PLAIN_SCALAR@65..66 "d"
        WHITESPACE@66..67 "\n"
        BLOCK_MAP_ENTRY@67..108
          BLOCK_MAP_KEY@67..83
            QUESTION_MARK@67..68 "?"
            WHITESPACE@68..69 " "
            COMMENT@69..75 "# lala"
            WHITESPACE@75..77 "\n "
            BLOCK@77..83
              BLOCK_SEQ@77..83
                BLOCK_SEQ_ENTRY@77..83
                  MINUS@77..78 "-"
                  WHITESPACE@78..79 " "
                  FLOW@79..83
                    PLAIN_SCALAR@79..83 "seq1"
          WHITESPACE@83..84 "\n"
          COLON@84..85 ":"
          WHITESPACE@85..86 " "
          COMMENT@86..92 "# lala"
          WHITESPACE@92..94 "\n "
          BLOCK_MAP_VALUE@94..108
            BLOCK@94..108
              BLOCK_SEQ@94..108
                BLOCK_SEQ_ENTRY@94..108
                  MINUS@94..95 "-"
                  WHITESPACE@95..96 " "
                  COMMENT@96..101 "#lala"
                  WHITESPACE@101..104 "\n  "
                  FLOW@104..108
                    PLAIN_SCALAR@104..108 "seq2"
        WHITESPACE@108..109 "\n"
        BLOCK_MAP_ENTRY@109..132
          BLOCK_MAP_KEY@109..110
            FLOW@109..110
              PLAIN_SCALAR@109..110 "e"
          COLON@110..111 ":"
          WHITESPACE@111..112 " "
          BLOCK_MAP_VALUE@112..132
            BLOCK@112..132
              PROPERTIES@112..117
                ANCHOR_PROPERTY@112..117
                  AMPERSAND@112..113 "&"
                  ANCHOR_NAME@113..117 "node"
              WHITESPACE@117..118 " "
              COMMENT@118..124 "# lala"
              WHITESPACE@124..126 "\n "
              BLOCK_SEQ@126..132
                BLOCK_SEQ_ENTRY@126..132
                  MINUS@126..127 "-"
                  WHITESPACE@127..128 " "
                  BLOCK@128..132
                    BLOCK_MAP@128..132
                      BLOCK_MAP_ENTRY@128..132
                        BLOCK_MAP_KEY@128..129
                          FLOW@128..129
                            PLAIN_SCALAR@128..129 "x"
                        COLON@129..130 ":"
                        WHITESPACE@130..131 " "
                        BLOCK_MAP_VALUE@131..132
                          FLOW@131..132
                            PLAIN_SCALAR@131..132 "y"
        WHITESPACE@132..133 "\n"
        BLOCK_MAP_ENTRY@133..156
          BLOCK_MAP_KEY@133..138
            FLOW@133..138
              PLAIN_SCALAR@133..138 "block"
          COLON@138..139 ":"
          WHITESPACE@139..140 " "
          BLOCK_MAP_VALUE@140..156
            BLOCK@140..156
              BLOCK_SCALAR@140..156
                GREATER_THAN@140..141 ">"
                WHITESPACE@141..142 " "
                COMMENT@142..148 "# lala"
                BLOCK_SCALAR_TEXT@148..156 "\n  abcde"
  WHITESPACE@156..157 "\n"
